test/hypothesis_utils.py

```diff
@@ -182,6 +182,38 @@ def tensor(draw, shapes=None, elements=None, qparams=None):
         zp = enforced_zp
     return X, (scale, zp, qparams[2])
 
+@st.composite
+def per_channel_tensor(draw, shapes=None, elements=None, qparams=None):
+    if isinstance(shapes, SearchStrategy):
+        _shape = draw(shapes)
+    else:
+        _shape = draw(st.sampled_from(shapes))
+    if qparams is None:
+        if elements is None:
+            elements = st.floats(-1e6, 1e6, allow_nan=False)
+        X = draw(stnp.arrays(dtype=np.float32, elements=elements, shape=_shape))
+        assume(not (np.isnan(X).any() or np.isinf(X).any()))
+        return X, None
+    qparams = draw(qparams)
+    if elements is None:
+        min_value, max_value = _get_valid_min_max(qparams)
+        elements = st.floats(min_value, max_value, allow_infinity=False,
+                             allow_nan=False)
+    X = draw(stnp.arrays(dtype=np.float32, elements=elements, shape=_shape))
+    # Recompute the scale and zero_points according to the X statistics.
```

Comment:
I don't really understand, why do we need to recalculate qparams?

Reply:
This is the same logic that is used for the per-tensor quantization, if we dont calculate qparams we end up with invalid choices of scale and zero-point that do not match the tensor. For example, you could end up with a tensor having a small dynamic range having a very large scale or even an invalid choice of zero-point.

Reply:
then we should change the tensor generation process to make sure zero_point is valid I think, generate tensors with some scale/zero_point and then recalculate the scale/zero_point again doesn't sound correct

Reply:
This is the same issue that @llyfacebook has been looking at. Let me address this in a seperate PR that fixes hypothesis_utils for both per-tensor and per-channel. (@llyfacebook : Do you have something in the works for this?)

Reply:
Currently, I think we still need to do recalculate. Otherwise, all the resulted matrix will be with same element given large scale.